### PR TITLE
Add anchor tags to `ManagedCard` component, update "Add new Destination" button to scroll user to "Destinations" card

### PR DESCRIPTION
**Pull request by rwfeather**
## Change description

This PR adds `id` fields to the `ManagedCard` header, making them linkable.

Now when a user clicks the "Add new Destination" button on setup steps, we take them to the model overview page with the destinations section scrolled into view.

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

**Comment by rwfeather** [comment]
Follow up story for getting page scrolling / position to work: https://www.pivotaltracker.com/story/show/181035427

**Review by krishnaglick — COMMENTED** [review]
Would it be possible to add a snapshot test of the loading state?

**Comment by rwfeather** [comment]
> Would it be possible to add a snapshot test of the loading state?

Yup, I'll add a test for that.

File: ui/ui-components/__tests__/components/managedCard.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { render } from "@testing-library/react";
+import ManagedCard from "../../components/lib/ManagedCard";
+
+describe("managedCard", () => {
+  it("will render a spinner when in the loading state", async () => {
+    let card = render(<ManagedCard title="test" loading={false}></ManagedCard>);
+    expect(card.container).toMatchSnapshot("noLoading");
+    card = render(<ManagedCard title="test" loading={true}></ManagedCard>);
+    expect(card.container).toMatchSnapshot("loading");
```
Comment (krishnaglick) [inline ui/ui-components/__tests__/components/managedCard.tsx:9]: This is great!
Minor note, you don't need to name snapshots if you don't want to, jest will handle it 👍

**Review by krishnaglick — APPROVED** [review]

**Pull request merged**